tutorial-registry/schema.json

```diff
@@ -0,0 +1,73 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://raw.githubusercontent.com/scverse/scverse-tutorials/main/schema.json",
+    "title": "Scverse Tutorials",
+    "description": "A tutorial for an scverse core or ecosystem package that is listed on scverse.org/learn",
+    "type": "object",
+    "properties": {
+        "name": {
+            "description": "The name of the tutorial. Must be unique.",
+            "type": "string"
+        },
+        "description": {
+            "description": "Summary of the tutorial (1-2 sentences).",
+            "type": "string"
+        },
+        "link": {
+            "description": "Link to the tutorial",
+            "type": "string",
+            "format": "uri"
+        },
+        "level": {
+            "description": "Difficulty of the tutorial",
+            "type": "string",
+            "enum": ["beginner", "intermediate", "advanced"]
+        },
+        "category": {
+            "description": "Category of the tutorial",
```

Comment:
Really tricky to nail the categories with the right resolution. Maybe add

`machine learning`

Why is `data integration` its own category?

Reply:
agreed. I don't think this list is final, were just a few terms that came to my mind.

Reply:
Do we need to start out with an enum here? I feel like we'd end up modifying the schema quite often

Reply:
I feel if we don't we end up with one category per tutorial :shrug: 
I also don't think it's a problem to add an additional category to the schema in the same PR adding a new tutorial using that category.

Reply:
ok, let's rename this to "topic" and allow multiple values. 

I would still be in favor of an enum that is adapted dynamically. Without, it's hard to know which topics already exist and similar ones will be created. 

This is what we have now as tags for ecosystem packages, which is pretty much useless: 
```
Counter({'GRN': 1,
         'TF': 3,
         'ML': 1,
         'cell-fate': 2,
         'rna-velocity': 1,
         'trajectory-generation': 2,
         'BLAST': 1,
         'visualization': 1,
         'doublet': 1,
         'single cell': 2,
         'optimal transport': 1,
         'multi omics': 1,
         'data integration': 2,
         'NMF': 1,
         'pathology': 1,
         'rna-seq': 1,
         'differential-expression': 1,
         'SnapATAC': 1,
         'R package': 1,
         'spatial analysis': 1,
         'single-cell': 2,
         'disentanglement': 1,
         'generative framework': 1,
         'Bayesian': 1,
         'HCA': 1,
         'dandelion': 1,
         'bcr': 1,
         'tcr': 1,
         'GSEA': 1,
         'Pathway': 1,
         'vector': 1,
         'scanpy': 1,
         'epigenomics': 1,
         'gene-signatures': 1,
         'CNV': 1,
         'transcriptome analysis': 1,
         'regulatory networks': 1,
         'clustering': 2,
         'scATAC-seq': 1,
         'pseudotime': 1,
         'perturbation': 1,
         'benchmarking': 1,
         'rna velocity': 1,
         'data repository': 1,
         'signals': 1,
         'filter': 1,
         'spatial alignment': 1,
         'spatial registration': 1,
         'spatial decomposition': 1,
         'spatial mapping': 1,
         'imaging': 1})
```

Reply:
My initial idea was to have one "category" per tutorial, where each category has some descriptive text (see `categories.yml`) that is used to group the tutorials on the website, similar to how this is done on the corresponding [Seurat page](https://satijalab.org/seurat/articles/get_started.html).

E.g. "data integration" would be such a category in the following screenshot: 

![image](https://user-images.githubusercontent.com/7051479/232761009-ee5896f0-b3b8-4539-8fa8-b14f4afdecc9.png)

Reply:
Yeah, I think that this is great and makes sense

Reply:
> My initial idea was to have one "category" per tutorial, where each category has some descriptive text (see categories.yml) that is used to group the tutorials on the website

That makes sense to me, but then is this the source that generates the learn page? E.g., the `tutorials` yaml entries here: 

https://github.com/scverse/scverse.github.io/blob/06b3b34f3dba223b06eacda57c085b403cb2e016/content/learn/_index.md?plain=1#L11-L20

Reply:
>  but then is this the source that generates the learn page

Yes, that would be the plan. Same principle as for the ecosystem packages page.